### PR TITLE
Added documentation for report API

### LINKED ISSUE #6363: Update documentation related to transactions and authorizations API endpoint
### Description
Updated the documentation related to work done in #6336  and #6342

### PR DESCRIPTION
**Pull request by zmaglica**
Fixes #6363 

#### Changes proposed in this Pull Request

This PR adds documentation for reports API endpoints . Date filters for authorizations API endpoint will not be included into this PR. This is because there is an issue with them (sometimes they can return incorrect data) (https://github.com/Automattic/woocommerce-payments/issues/7445).

#### Testing instructions

- After merge, make sure that you can see the docs here: https://automattic.github.io/woocommerce-payments/#introduction
- Test added API endpoints and compare parameters and responses schema, do they match and is something is missing.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7320` or branch name `add/6363-add-documentation-for-reporting-api` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/6363-add-documentation-for-reporting-api) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 589e05be544f26198fc861e805c83c88f607fc19
- Build time: 2023-10-26 10:36:10 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by jessy-p — COMMENTED** [review]
Missing the links to the documentation from sidebar

**Review by naman03malhotra — COMMENTED** [review]
LGTM! 🚢 

Just want to confirm that the sidebar item will be picked automatically by `layout.erb`?

**Comment by zmaglica** [comment]
> Missing the links to the documentation from sidebar

Addressed here: https://github.com/Automattic/woocommerce-payments/pull/7320/commits/93c48b70eba4ff952774d8a69cc72b788a03c6ad

**Comment by zmaglica** [comment]
> LGTM! 🚢
> 
> Just want to confirm that the sidebar item will be picked automatically by `layout.erb`?

Sadly, it won't. I have pushed a new commit (https://github.com/Automattic/woocommerce-payments/commit/93c48b70eba4ff952774d8a69cc72b788a03c6ad) that adds sidebar link

File: docs/rest-api/source/includes/wp-api-v3/reports.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+		<i class="label label-get">GET</i>
+		<h6>/wp-json/wc/v3/payments/reports/authorizations</h6>
+	</div>
+</div>
+
+### Optional parameters
+
+- `date_before`: Filter authorizations before this date.
+- `date_after`: Filter authorizations after this date.
+- `date_between`: Filter authorizations between these dates.
```
Comment (naman03malhotra) [inline docs/rest-api/source/includes/wp-api-v3/reports.md]: @zmaglica can you please move this information to the date filter issue https://github.com/Automattic/woocommerce-payments/issues/7445 and merge this PR?

**Review by naman03malhotra — DISMISSED** [review]

File: docs/rest-api/source/includes/wp-api-v3/reports.md (hunk 1 above)
Comment (zmaglica) [reply docs/rest-api/source/includes/wp-api-v3/reports.md]: Addresed here: https://github.com/Automattic/woocommerce-payments/pull/7320/commits/bae478a7c82f215719aa54b55c5ca329afd0f66f

File: docs/rest-api/source/includes/wp-api-v3/reports.md [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+		<i class="label label-get">GET</i>
+		<h6>/wp-json/wc/v3/payments/reports/transactions</h6>
+	</div>
+</div>
+
+### Optional parameters
+
+- `date_before`: Filter transactions before this date.
+- `date_after`: Filter transactions after this date.
+- `date_between`: Filter transactions between these dates.
```
Comment (naman03malhotra) [inline docs/rest-api/source/includes/wp-api-v3/reports.md:29]: Is date filter for transactions is working?

**Comment by naman03malhotra** [comment]
Also, can you please link the new issue for context and the mention the place you've moved the work to add the date filters later?

File: docs/rest-api/source/includes/wp-api-v3/reports.md (hunk 2 above)
Comment (zmaglica) [reply docs/rest-api/source/includes/wp-api-v3/reports.md:29]: Yes, they do. I just tested it.

**Comment by zmaglica** [comment]
> Also, can you please link the new issue for context and the mention the place you've moved the work to add the date filters later?

In the PR description, I have referenced an issue where it is mentioned that date filters needs to be added to the authorizations API documentation once the problem with date filters itself is resolved.

**Comment by naman03malhotra** [comment]
The part for updating date filter documentation has been moved to https://github.com/Automattic/woocommerce-payments/issues/7445

**Review by naman03malhotra — APPROVED** [review]
LGTM! 🚢

**Pull request merged**